allennlp_models/rc/dataset_readers/transformer_squad.py

```diff
@@ -250,7 +294,10 @@ def text_to_instance(
                 token_answer_span[1] + start_of_context,
                 question_field,
             )
-        else:
+        elif for_training and self.use_cls_token_for_unanswerable:
+            cls_index = fields["cls_index"].sequence_index
+            fields["answer_span"] = SpanField(cls_index, cls_index, question_field)
+        elif for_training:
             # We have to put in something even when we don't have an answer, so that this instance can be batched
             # together with other instances that have answers.
             fields["answer_span"] = SpanField(-1, -1, question_field)
```

Comment:
Can't we do this all the time, and not have the extra flag at all?

Reply:
If we did this all the time, the model would end up computing a loss even when you're just trying to get predictions.

Reply:
Is that a problem?

It would not create a loss when we just don't have an answer.

Reply:
`answer_span` is the answer though.